### PR TITLE
Rails 7.2 support

### PR DESCRIPTION
**Pull request by MrAsler**
Rails 7.2 support
I pretty much copied the logic from the previous [rails update ](https://github.com/salsify/rails-multitenant/pull/44)

prime: @jturkel @erikkessler1

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -72,6 +72,7 @@ workflows:
                 - gemfiles/rails_6.1.gemfile
                 - gemfiles/rails_7.0.gemfile
                 - gemfiles/rails_7.1.gemfile
+                - gemfiles/rails_7.2.gemfile
```
Comment (jturkel) [inline .circleci/config.yml:75]: Looks like Rails 7.2 doesn't support Ruby 3.0. Ruby 3.0 is EOL so let's drop support for it. Rails 6.1 is also EOL so we can drop support for it too.

File: .circleci/config.yml [hunk 2]
```diff
@@ -72,8 +72,8 @@ workflows:
...
               ruby_version:
```
Comment (jturkel) [inline .circleci/config.yml:76]: Can we use latest maintenance release for each version of Ruby?

Comment (MrAsler) [reply .circleci/config.yml:75]: Should I also update each rails version in the Appraisals file to the latest maintenance release?

Comment (jturkel) [reply .circleci/config.yml:75]: That would be great. Thanks!

**Review by erikkessler1 — APPROVED** [review]

**Pull request merged**